File: webui/src/test/java/com/taskadapter/web/uiapi/TestUIConfigStoreFactory.java
package com.taskadapter.web.uiapi;

import com.taskadapter.PluginManager;
import com.taskadapter.auth.cred.FSCredentialStore;
import com.taskadapter.config.ConfigStorage;
import com.taskadapter.webui.service.EditorManager;

import java.io.File;

public class TestUIConfigStoreFactory {
    public static UIConfigStore createStore(File temporaryFolder) {
        var configStorage = new ConfigStorage(temporaryFolder);
        var editorManager = EditorManager.fromResource("editors.txt");
        var uiConfigService = new UIConfigService(new PluginManager(), editorManager);
        return new UIConfigStore(uiConfigService, configStorage, new FSCredentialStore(temporaryFolder));
    }

}

File: webui/src/test/scala/com/taskadapter/webui/uiapi/TestUIConfigStoreFactory.scala
package com.taskadapter.webui.uiapi

import java.io.File

import com.taskadapter.PluginManager
import com.taskadapter.config.ConfigStorage
import com.taskadapter.web.uiapi.{UIConfigService, UIConfigStore}
import com.taskadapter.webui.service.EditorManager

object TestUIConfigStoreFactory {
  def createStore(temporaryFolder: File): UIConfigStore = {
    val configStorage = new ConfigStorage(temporaryFolder)
    val editorManager = EditorManager.fromResource("editors.txt")
    val uiConfigService = new UIConfigService(new PluginManager, editorManager)
    new UIConfigStore(uiConfigService, configStorage)
  }
}
